pkg/machinery/config/types/v1alpha1/v1alpha1_validation.go

```diff
@@ -480,26 +488,83 @@ func checkWireguard(b *DeviceWireguardConfig) error {
 	return result.ErrorOrNil()
 }
 
+func checkVlans(d *Device) error {
+	var result *multierror.Error
+
+	// check VLAN addressing
+	for _, vlan := range d.DeviceVlans {
+		if len(vlan.VlanAddresses) > 0 && vlan.VlanCIDR != "" {
+			result = multierror.Append(result, fmt.Errorf("[%s] %s.%d: %s", "networking.os.device.vlan", d.DeviceInterface, vlan.VlanID, "vlan can't have both .cidr and .addresses set"))
+		}
+
+		if vlan.VlanCIDR != "" {
+			if err := validateIPOrCIDR(vlan.VlanCIDR); err != nil {
+				result = multierror.Append(result, fmt.Errorf("[%s] %s.%d: %w", "networking.os.device.vlan.CIDR", d.DeviceInterface, vlan.VlanID, err))
+			}
+		}
+
+		for _, address := range vlan.VlanAddresses {
+			if err := validateIPOrCIDR(address); err != nil {
+				result = multierror.Append(result, fmt.Errorf("[%s] %s.%d: %w", "networking.os.device.vlan.addresses", d.DeviceInterface, vlan.VlanID, err))
+			}
+		}
+	}
+
+	return result.ErrorOrNil()
+}
+
+func validateIPOrCIDR(address string) error {
```

Comment:
This is the same routine as the `parseIPorPrefix` above, just discarding the value.  Perhaps pull it out into net tools or something?

Reply:
there's a small difference though: this function uses `net` package, while `parseIPOrPrefix` uses `netaddr`, I was trying not to introduce `netaddr` dependency in the `machinery` package

Reply:
Good point; works for me.